src/convolution/u16x1/wasm32.rs

```diff
@@ -97,16 +97,16 @@ unsafe fn horiz_convolution_four_rows(
                 let source = wasm32_utils::load_v128(src_rows[i], x);
 
                 let l0l1_i64x2 = i8x16_swizzle(source, L0L1_SHUFFLE);
-                sum = i64x2_add(sum, i64x2_mul(l0l1_i64x2, coeff01_i64x2));
+                sum = i64x2_add(sum, wasm32_utils::i64x2_mul_lo(l0l1_i64x2, coeff01_i64x2));
```

Comment:
`i64x2_mul_lo()` does additional shuffles of data that already was shuffled. Maybe it would be better if you prepare the data in the right form in advance?

Reply:
You may write something like this:
```rust
let l0l1l2l3_i32x4 = i8x16_swizzle(source, L0L1L2L3_SHUFFLE);
sum = i64x2_add(sum, i64x2_extmul_low_i32x4(l0l1l2l3_i32x4, coeff0123_i32x4));
sum = i64x2_add(sum, i64x2_extmul_high_i32x4(l0l1l2l3_i32x4, coeff0123_i32x4));
```

Reply:
Yes. I did that but for some reason it was much slower. I'm still trying to figure out why.

Reply:
I've found the issue, but I can't figure out a solution. By changing the shuffle values to avoid the extra shuffles, the code is optimized to use the very inefficient functions `i64x2_mul`, `i64x2_shr_s`, and `i64x2_shl`.
https://emscripten.org/docs/porting/simd.html
```
da =  
          ((((i64x2_extend_low_i32x4_s(ka i8x16_swizzle (la = V128)) i64x2_mul
              (ma = (SimdShuffleOp(LoadSplat(d), ka) i64x2_shl 32) i64x2_shr_s 32)) i64x2_add
             da) i64x2_add
            (i64x2_extend_low_i32x4_s(ka i8x16_swizzle (na = V128)) i64x2_mul
             (oa =
                (SimdShuffleOp(LoadSplat(d + 4), ka) i64x2_shl 32) i64x2_shr_s 32))) i64x2_add
           (i64x2_extend_low_i32x4_s(ka i8x16_swizzle (pa = V128)) i64x2_mul
            (qa =
               (SimdShuffleOp(LoadSplat(d + 8), ka) i64x2_shl 32) i64x2_shr_s 32))) i64x2_add
          (i64x2_extend_low_i32x4_s(ka i8x16_swizzle (ra = V128)) i64x2_mul
           (ka =
              (SimdShuffleOp(LoadSplat(d + 12), ka) i64x2_shl 32) i64x2_shr_s 32));
```
The code in this PR is compiled to
```
da = 
          ((((((ka i8x16_swizzle (la = V128)) i8x16_swizzle (ma = V128)) i64x2_extmul_low_i32x4_s
              (na = i64x2_splat(d[0]:int) i8x16_swizzle ma)) i64x2_add
             da) i64x2_add
            (((ka i8x16_swizzle (oa = V128)) i8x16_swizzle ma) i64x2_extmul_low_i32x4_s
             (pa = i64x2_splat((d + 4)[0]:int) i8x16_swizzle ma))) i64x2_add
           (((ka i8x16_swizzle (qa = V128)) i8x16_swizzle ma) i64x2_extmul_low_i32x4_s
            (ra = i64x2_splat((d + 8)[0]:int) i8x16_swizzle ma))) i64x2_add
          (((ka i8x16_swizzle (sa = V128)) i8x16_swizzle ma) i64x2_extmul_low_i32x4_s
           (ka = i64x2_splat((d + 12)[0]:int) i8x16_swizzle ma));```